cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -217,14 +217,19 @@ public void sanityCheckDryRun(boolean dryRun, boolean stopOngoingExecution) {
     if (dryRun) {
       return;
     }
-    if (_executor.hasOngoingExecution() && !stopOngoingExecution) {
-      throw new IllegalStateException(String.format("Cannot start a new execution while there is an ongoing execution. "
-                                                    + "Please use %s=true to stop ongoing execution and start a new one.",
-                                                    STOP_ONGOING_EXECUTION_PARAM));
-    } else if (_executor.hasOngoingPartitionReassignments()) {
-      throw new IllegalStateException("Cannot execute new proposals while there are ongoing partition reassignments.");
-    } else if (_executor.hasOngoingLeaderElection()) {
-      throw new IllegalStateException("Cannot execute new proposals while there are ongoing leadership reassignments.");
+    if (_executor.hasOngoingExecution()) {
+      if (!stopOngoingExecution) {
+        throw new IllegalStateException(String.format("Cannot start a new execution while there is an ongoing execution. "
+                                                      + "Please use %s=true to stop ongoing execution and start a new one.",
+                                                      STOP_ONGOING_EXECUTION_PARAM));
+      }
+    } else {
+      if (_executor.hasOngoingPartitionReassignments()) {
+        throw new IllegalStateException("Cannot execute new proposals while there are ongoing partition reassignments.");
+      }
+      if (_executor.hasOngoingLeaderElection()) {
```

Comment:
Nit: Is the change from `else if (_executor.hasOngoingLeaderElection())` to `if (_executor.hasOngoingLeaderElection())` needed?